RecoLocalTracker/SiPixelRecHits/plugins/PixelRecHits.cu

```diff
@@ -59,27 +89,50 @@ namespace pixelgpudetails {
       input.nDigis,
       gpu_.hitsModuleStart_d,
       gpu_.charge_d,
-      gpu_.xg_d, gpu_.yg_d, gpu_.zg_d,
-      gpu_.xerr_d, gpu_.yerr_d, gpu_.mr_d,
-      true // for the time being stay local...
+      gpu_.detInd_d,
+      gpu_.xg_d, gpu_.yg_d, gpu_.zg_d, gpu_.rg_d,
+      gpu_.iphi_d,
+      gpu_.xl_d, gpu_.yl_d,
+      gpu_.xerr_d, gpu_.yerr_d, 
+      gpu_.mr_d, gpu_.mc_d
     );
 
     // needed only if hits on CPU are required...
     cudaCheck(cudaMemcpyAsync(hitsModuleStart_, gpu_.hitsModuleStart_d, (gpuClustering::MaxNumModules+1) * sizeof(uint32_t), cudaMemcpyDefault, stream.id()));
+   
+    // to be moved to gpu?
+    auto nhits = hitsModuleStart_[gpuClustering::MaxNumModules];
+    for (int i=0;i<10;++i) hitsLayerStart_[i]=hitsModuleStart_[phase1PixelTopology::layerStart[i]];
+    hitsLayerStart_[10]=nhits;
+
+    std::cout << "hit layerStart "; 
+    for (int i=0;i<10;++i) std::cout << phase1PixelTopology::layerName[i] << ':' << hitsLayerStart_[i] << ' ';
+    std::cout << "end:" << hitsLayerStart_[10] << std::endl;
```

Comment:
By the way, now we have a per-event (non-thread safe) printout from here.

Reply:
will be removed at next iteration when the sorting will be implemented as well